src/app/hooks/portfolio/useConvertToXUSD.ts

```diff
@@ -0,0 +1,43 @@
+import { useAccount } from '../useAccount';
+import { useSendContractTx } from '../useSendContractTx';
+import { getContract } from 'utils/blockchain/contract-helpers';
+import { TxType } from '../../../store/global/transactions-store/types';
+import { gasLimit } from 'utils/classifiers';
+import {
+  CheckAndApproveResult,
+  contractWriter,
+} from 'utils/sovryn/contract-writer';
+import { Asset } from 'types/asset';
+
+export const useConvertToXUSD = () => {
+  const account = useAccount();
+  const rusdtAddress = getContract('USDT_token').address;
+  const { send, ...rest } = useSendContractTx('babelfishAggregator', 'mintTo');
+
+  return {
+    convert: async (weiAmount: string) => {
+      let tx: CheckAndApproveResult = {};
+
+      tx = await contractWriter.checkAndApprove(
+        Asset.USDT,
+        getContract('USDT_token').address,
```

Comment:
Right now you are approving USDT token to be used by USDT token, spender must be babelfishAggregator here:

```suggestion
        getContract('babelfishAggregator').address,
```